DEC/src/theories/ThDEC/semantics.jl

```diff
@@ -0,0 +1,73 @@
+import Decapodes
+using StructEquality
+
+
+"""    create_dynamic_model(sd, hodge)::Dict{TypedApplication, Any}
+
+Given a matrix and a hodge star (DiagonalHodge() or GeometricHodge()), this returns a lookup dictionary between operators (as TypedApplications) and their corresponding matrices.
+
+"""
+function create_dynamic_model(sd, hodge)::Dict{TypedApplication, Any}
```

Comment:
Purposes of using a Dict here vs. an `@match` block?

Reply:
I think the blessed way to do this is going to actually be to create a memoized function/lazy dict. Best of both worlds: we don't have to compute operators that we don't need, and we don't have to compute operators multiple times.